server/src/test/java/org/opensearch/action/admin/indices/tiering/TransportHotToWarmTieringActionTests.java

```diff
@@ -0,0 +1,118 @@
+        featureSettings.put(FeatureFlags.TIERED_REMOTE_INDEX, true);
+        return featureSettings.build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Collections.singletonList(MockInternalClusterInfoService.TestPlugin.class);
+    }
+
+    @Before
+    public void setup() {
+        internalCluster().startClusterManagerOnlyNode();
+        internalCluster().ensureAtLeastNumSearchAndDataNodes(1);
+        long bytes = new ByteSizeValue(1000, ByteSizeUnit.KB).getBytes();
+        final MockInternalClusterInfoService clusterInfoService = getMockInternalClusterInfoService();
+        clusterInfoService.setDiskUsageFunctionAndRefresh((discoveryNode, fsInfoPath) -> setDiskUsage(fsInfoPath, bytes, bytes - 1));
+
+        final int numReplicasIndex = 0;
+        final Settings settings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, numReplicasIndex)
+            .build();
+
+        indices = new String[] { TEST_IDX_1, TEST_IDX_2 };
+        for (String index : indices) {
+            assertAcked(client().admin().indices().prepareCreate(index).setSettings(settings).get());
+            ensureGreen(index);
+        }
+    }
+
+    @After
+    public void cleanup() {
+        client().admin().indices().prepareDelete(indices).get();
+    }
+
+    MockInternalClusterInfoService getMockInternalClusterInfoService() {
+        return (MockInternalClusterInfoService) internalCluster().getCurrentClusterManagerNodeInstance(ClusterInfoService.class);
+    }
+
+    static FsInfo.Path setDiskUsage(FsInfo.Path original, long totalBytes, long freeBytes) {
+        return new FsInfo.Path(original.getPath(), original.getMount(), totalBytes, freeBytes, freeBytes);
+    }
+
+    public void testIndexLevelBlocks() {
+        enableIndexBlock(TEST_IDX_1, SETTING_READ_ONLY_ALLOW_DELETE);
+        TieringIndexRequest request = new TieringIndexRequest(TARGET_TIER, TEST_IDX_1);
+        expectThrows(ClusterBlockException.class, () -> client().execute(HotToWarmTieringAction.INSTANCE, request).actionGet());
+    }
+
+    public void testIndexNotFound() {
+        TieringIndexRequest request = new TieringIndexRequest(TARGET_TIER, "foo");
+        expectThrows(IndexNotFoundException.class, () -> client().execute(HotToWarmTieringAction.INSTANCE, request).actionGet());
+    }
+
+    public void testNoConcreteIndices() {
+        TieringIndexRequest request = new TieringIndexRequest(TARGET_TIER, "foo");
+        request.indicesOptions(IndicesOptions.fromOptions(true, true, true, false));
+        HotToWarmTieringResponse response = client().admin().indices().execute(HotToWarmTieringAction.INSTANCE, request).actionGet();
+        assertFalse(response.isAcknowledged());
+        assertTrue(response.getFailedIndices().isEmpty());
+    }
+
+    public void testNoAcceptedIndices() {
```

Comment:
I think the test case for good case when there is accepted indices will come in the PR with service side of changes ?

Reply:
yes thats correct